crates/signer/src/manager/dirk.rs

```diff
@@ -0,0 +1,917 @@
+            "Failed to add new proxy account to accounts map".to_string(),
+        ))?;
+        trace!(?hashmap_key, ?added, "Proxy account added");
+
+        // Get delegation signature from the consensus account
+        let message = ProxyDelegation { delegator, proxy: proxy_key };
+        let signature = self.request_signature(delegator, message.tree_hash_root().0).await?;
+        let delegation: SignedProxyDelegation<BlsPublicKey> =
+            SignedProxyDelegation { message, signature };
+
+        if let Some(store) = &self.proxy_store {
+            store.store_proxy_bls_delegation(&module_id, &delegation).map_err(|err| {
+                SignerModuleError::Internal(format!("error storing delegation signature: {err}"))
+            })?;
+        }
+
+        Ok(delegation)
+    }
+
+    pub async fn request_signature(
+        &self,
+        pubkey: BlsPublicKey,
+        object_root: [u8; 32],
+    ) -> Result<BlsSignature, SignerModuleError> {
+        let domain = compute_domain(self.chain, COMMIT_BOOST_DOMAIN);
+        let key = hex::encode(pubkey);
+        let account = self
+            .accounts
+            .get(&key)
+            .ok_or_else(|| SignerModuleError::UnknownConsensusSigner(pubkey.to_vec()))?;
+
+        match account.wallet_type {
+            WalletType::Simple => {
+                // No need for extra logic since we're just signing to a single channel
+                let channel = self.get_channel_for_pubkey(&pubkey)?;
+                self.sign_with_channel(channel, pubkey, account, domain, object_root).await
+            }
+            WalletType::Distributed => {
+                let mut signatures = Vec::new();
+                let mut identifiers = Vec::new();
+                let num_hosts_needed = account.signing_threshold as usize;
+
+                if account.hosts.len() < num_hosts_needed {
+                    return Err(SignerModuleError::Internal(format!(
+                        "Not enough hosts available. Need {} but only have {}",
+                        num_hosts_needed,
+                        account.hosts.len()
+                    )));
+                }
+
+                // Try to get signatures from hosts
+                for host in &account.hosts {
+                    if signatures.len() >= num_hosts_needed {
+                        break;
+                    }
+
+                    let channel = self.channels.get(&host.domain).cloned().ok_or_else(|| {
+                        SignerModuleError::Internal(format!(
+                            "No channel found for host {}",
+                            host.domain
+                        ))
+                    })?;
+
+                    trace!(host = host.domain, "Requesting signature shard for creating proxy");
+                    match self
+                        .sign_with_channel(channel, pubkey, account, domain, object_root)
+                        .await
```

Comment:
We should request all signatures and then wait for all (or at least the threshold number)

Reply:
Done in 215ed7039adb89bfeb733ad521eb40101f39842c , from what I've checked it seems to be working fine.